The PR is failing due to the increase in the number of warnings, apparently there are a few added in this PR: 
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:13: warning: main actor-isolated property '\''settingScreen'\'' can not be mutated from a nonisolated context
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:29: warning: call to main actor-isolated initializer '\''init(app:selectors:)'\'' in a synchronous nonisolated context'
```
+ envman add --key SHOW_WARNING_IN_SLACK --value Show_Warnings
+ envman add --key SHOW_WARNING_COUNT_XCUI --value 'Number of warnings is:        5. This is greater than build threshold: 3

--- Warnings ---
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1625:16: warning: passing closure as a '\''sending'\'' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1593:16: warning: passing closure as a '\''sending'\'' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/firefox-ios/Client/Extensions/UIDevice+Extension.swift:29:17: warning: '\''init(validatingUTF8:)'\'' is deprecated: replaced by '\''String.init(validatingCString:)'\''
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:13: warning: main actor-isolated property '\''settingScreen'\'' can not be mutated from a nonisolated context
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:29: warning: call to main actor-isolated initializer '\''init(app:selectors:)'\'' in a synchronous nonisolated context'
+ STR='Number of warnings is:        5. This is greater than build threshold: 3

--- Warnings ---
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1625:16: warning: passing closure as a '\''sending'\'' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1593:16: warning: passing closure as a '\''sending'\'' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/firefox-ios/Client/Extensions/UIDevice+Extension.swift:29:17: warning: '\''init(validatingUTF8:)'\'' is deprecated: replaced by '\''String.init(validatingCString:)'\''
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:13: warning: main actor-isolated property '\''settingScreen'\'' can not be mutated from a nonisolated context
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:29: warning: call to main actor-isolated initializer '\''init(app:selectors:)'\'' in a synchronous nonisolated context'
+ SUB=greater
+ [[ Number of warnings is:        5. This is greater than build threshold: 3

--- Warnings ---
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1625:16: warning: passing closure as a 'sending' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/MozillaRustComponents/Sources/MozillaRustComponentsWrapper/Generated/viaduct.swift:1593:16: warning: passing closure as a 'sending' parameter risks causing data races between code in the current task and concurrent execution of the closure; this is an error in the Swift 6 language mode
/Users/vagrant/git/firefox-ios/Client/Extensions/UIDevice+Extension.swift:29:17: warning: 'init(validatingUTF8:)' is deprecated: replaced by 'String.init(validatingCString:)'
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:13: warning: main actor-isolated property 'settingScreen' can not be mutated from a nonisolated context
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:29: warning: call to main actor-isolated initializer 'init(app:selectors:)' in a synchronous nonisolated context == *\g\r\e\a\t\e\r* ]]
+ echo 'Failure, the number of warnings increased'
Failure, the number of warnings increased
```

These are the two new: 
```
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:13: warning: main actor-isolated property '\''settingScreen'\'' can not be mutated from a nonisolated context
/Users/vagrant/git/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift:15:29: warning: call to main actor-isolated initializer '\''init(app:selectors:)'\'' in a synchronous nonisolated context'
```

Could you please address those @sushanttiwari-14 , let us know if you need any help

--- a/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift
+++ b/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift
@@ -9,7 +9,6 @@ let defaultSearchEngine2 = "Bing"
 let customSearchEngine = ["name": "youtube", "url": "https://youtube.com/search?q=%s"]
 
 class SearchSettingsUITests: BaseTestCase {
-
     // https://mozilla.testrail.io/index.php?/cases/view/2435664
     func testDefaultSearchEngine() {
         let settingScreen = SettingScreen(app: app)

--- a/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift
+++ b/firefox-ios/firefox-ios-tests/Tests/XCUITests/SearchSettingsUITest.swift
@@ -9,13 +9,10 @@ let defaultSearchEngine2 = "Bing"
 let customSearchEngine = ["name": "youtube", "url": "https://youtube.com/search?q=%s"]
 
 class SearchSettingsUITests: BaseTestCase {
-    private var settingScreen: SettingScreen!
-    override func setUp() {
-            super.setUp()
-            settingScreen = SettingScreen(app: app)
-    }
+
     // https://mozilla.testrail.io/index.php?/cases/view/2435664
     func testDefaultSearchEngine() {
+        let settingScreen = SettingScreen(app: app)
         settingScreen.navigateToSearchSettings()
         // Check the default browser
         let defaultSearchEngine = app.tables.cells.element(boundBy: 0)
@@ -30,6 +27,7 @@ class SearchSettingsUITests: BaseTestCase {
 
     // https://mozilla.testrail.io/index.php?/cases/view/2353247
     func testCustomSearchEngineIsEditable() {
+        let settingScreen = SettingScreen(app: app)
         settingScreen.navigateToSearchSettings()
         // Add a custom search engine
         addCustomSearchEngine()
@@ -60,6 +58,7 @@ class SearchSettingsUITests: BaseTestCase {
 
     // https://mozilla.testrail.io/index.php?/cases/view/2353248
     func testCustomSearchEngineAsDefaultIsNotEditable() {
+        let settingScreen = SettingScreen(app: app)
         settingScreen.navigateToSearchSettings()
         // Edit is disabled
         XCTAssertFalse(app.buttons["Edit"].isEnabled)
@@ -79,6 +78,7 @@ class SearchSettingsUITests: BaseTestCase {
 
     // https://mozilla.testrail.io/index.php?/cases/view/2353249
     func testNavigateToSearchPickerTurnsOffEditing() {
+        let settingScreen = SettingScreen(app: app)
         settingScreen.navigateToSearchSettings()
         // Edit is disabled
         XCTAssertFalse(app.buttons["Edit"].isEnabled)
@@ -104,6 +104,7 @@ class SearchSettingsUITests: BaseTestCase {
 
     // https://mozilla.testrail.io/index.php?/cases/view/2353250
     func testDeletingLastCustomEngineExitsEditing() {
+        let settingScreen = SettingScreen(app: app)
         settingScreen.navigateToSearchSettings()
         // Edit is disabled
         XCTAssertFalse(app.buttons["Edit"].isEnabled)

